src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,151 @@
+     */
+    public void showLoadingError(String errorMessage) {
+        System.out.println("OOPS!! An error has occurred: " + errorMessage);
+    }
+
+    public boolean getIsExit() {
+        return this.isExit;
+    }
+
+    /**
+     * Inform user that task has been added.
+     * @param task task that was added.
+     * @param numTasks number of tasks in the user's tasklist.
+     */
+    public void showAdded(Task task, int numTasks) {
+        System.out.println("Got it. I've added this task:\n"
+                + task.toString());
+        System.out.printf("Now you have %d tasks in your list.\n",
+                numTasks);
+    }
+
+    /**
+     * Inform user that task has been deleted.
+     * @param deletedTask task that was deleted.
+     * @param numTask number of remaining tasks in the user's tasklist.
+     */
+    public void showDeleted(Task deletedTask, int numTask) {
+        System.out.println("Noted. I've removed this task: \n"
+                + deletedTask.toString());
+        System.out.printf("Now you have %d tasks in your list.\n",
+                numTask);
+    }
+
+    /**
+     * Inform user that task has already been marked as done.
+     * @param task task that was already done.
+     */
+    public void showTaskDone(Task task) {
+        System.out.println("This task has already been completed!\n"
+                + task.toString());
+    }
+
+    /**
+     * Inform user that task was marked as done.
+     * @param task task that was marked as done.
+     */
+    public void showMarkedDone(Task task) {
+        System.out.println("Nice! I've marked this task as done: \n"
+                + task.toString());
+    }
+
+    /**
+     * Displays the list of tasks in the user's TaskList when 'list' command is used.
+     * @param tasks user's TaskList.
+     */
+    public void showList(TaskList tasks) {
+        if (tasks.taskList.isEmpty()) {
+            System.out.println("There are no duke.tasks on your list. *quack*");
+        } else if (tasks.getLength() == 1) {
+            System.out.println("There is one task on your list:");
+            System.out.println("1. " + tasks.getTask(0).toString());
+            System.out.println("*quack*");
+        } else {
+            System.out.println("Here are the duke.tasks on your list:");
+            int i = 1;
+            for (Task task : tasks.taskList) {
+                System.out.println(i + ". " + task.toString());
+                i++;
+            }
+            System.out.println("*quack*\n");
+        }
+    }
+
+    public void showFindResults(ArrayList<Task> results) {
+        if (results.isEmpty()) {
+            System.out.println("Sorry, I did not find any task that matches your search *sad quack*\n");
+        } else {
+            System.out.println("Here are the tasks I found:");
+            for (Task task : results) {
+                System.out.println(task.toString());
+            }
+            System.out.println("*quack*\n");
+        }
```

Comment:
```suggestion
            
            System.out.println("Here are the tasks I found:");
            for (Task task : results) {
                System.out.println(task.toString());
            }
            System.out.println("*quack*\n");
```
I think code can look nicer and clearer without the else.